### ISSUE TITLE
Test that you can pin_read a public pin on RStudio Connect

### ISSUE THREAD

**Issue by machow** [rstudio/pins-python#146]
(Via a question from solutions engineering)

I think pins might assume you need an API key

**Comment by isabelizimm** [comment]
```
import pins
board = pins.board_rsconnect(
    server_url="https://colorado.rstudio.com/rsc", allow_pickle_read=True
)
board.pin_read("isabel.zimmerman/iz-cars")
```

returns 

```
pins.rsconnect.api.RsConnectApiRequestError: {'code': 24, 'error': 'The requested operation requires authentication.', 'payload': None}
```

where `isabel.zimmerman/iz-cars` [is public](https://colorado.rstudio.com/rsc/connect/#/apps/aa5ee310-bb02-471c-a7b1-5cf8ec75adb8/access)

**Comment by machow** [comment]
Hmm... I'm not sure pins can implement pin read without authentication. 

The error seems to occur when `board.pin_read("some_pin")` triggers:

* `pin_versions()` to get most recent version, which leads to...
* `pin_exists()` to ensure there's a pin to get versions from, which leads to...
* `<RsConnectFs>.exists()`, which has to convert `"some_pin"` to a guid, so
* makes a GET request to `/__api__/v1/users`

which you can't do without authenticating. If there were a way to convert `<user_name>/<content_name>` to a guid without authentication, then it might be possible?

**Comment by isabelizimm** [comment]
Running into place where I would like to read Connect publicly (for building model card + monitoring dashboard templates) again. Is there anything I can do to help this along?

**Comment by machow** [comment]
Can you provide example code and a traceback? Is `from_pin()` it erroring when it tries to `board.pin_versions(...)`?

I'm not sure I understand how vetiver's `_choose_version` differs from `board.pin_read(some_version, None)`. Is there a situation in which it does something different?

**Comment by machow** [comment]
Ah, I'm noticing you check for an "active" column -- which is what pin_versons returns for the RSConnect board in R, but this is not implemented in python apparently...

**R**

<img width="819" alt="image" src="https://user-images.githubusercontent.com/2574498/189201923-aace2e31-43e5-4db7-838b-4988b0c17702.png">

**Python**

<img width="891" alt="image" src="https://user-images.githubusercontent.com/2574498/189202066-44193a4b-835b-4831-b784-928e26cf3e78.png">

**Comment by machow** [comment]
Should be resolved in v0.0.6, which has a test of publicly reading a pin with board_url

**Comment by machow** [comment]
Here's an example translated from [this test](https://github.com/rstudio/pins-python/blob/main/pins/tests/test_boards.py#L447-L460)

```python
from pins import board_url, board_rsconnect

board_short = board_rsconnect(SOME_SERVER_URL)
board_short.pin_write(df, "susan/mtcars", type="csv", access_type="all")

# note that users can also get this from the web ui
content_url = board_short.fs.info("susan/mtcars")["content_url"]

board = board_url("", pin_paths={"rsc_public": content_url})

df_no_key = board.pin_read("rsc_public")
```
